velox/functions/sparksql/String.h

```diff
@@ -218,6 +218,28 @@ struct Sha2HexStringFunction {
   }
 };
 
+/// space function
+/// space(int) -> string
+/// Returns a string consisting of `n` spaces.
+template <typename T>
+struct SpaceFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const int32_t& count) {
+    if (count <= 0) {
+      result.setEmpty();
+      return;
+    }
+    result.append(generateSpaces(count));
```

Comment:
We can directly resize result to pre-allocate memory and then directly set the result buffer, i.e., `result.data()`.

Reply:
> We can directly resize result to pre-allocate memory and then directly set the result buffer, i.e., `result.data()`.

I modified it, please review again